account/src/main/java/com/user/account/controller/AllControlers.java

```diff
@@ -1,65 +1,69 @@
 package com.user.account.controller;
 
+import java.util.List;
+
+import com.user.account.advice.CustomExceptions;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
 
 import com.user.account.entity.User;
-import com.user.account.services.AccountServices;
-
+import com.user.account.message.DefaultMessage;
+import com.user.account.entity.Transactions;
+import com.user.account.services.AccountService;
+import com.user.account.services.TransactionService;
 
+import javax.validation.constraints.Min;
 
+@Validated
 @RestController
+@RequestMapping("/")
 public class AllControlers {
-	
-	
-		@Autowired
-		private AccountServices accountServices; 
-		
-		// tomcat test
-		@GetMapping("/home")
-		public String home() {
-			return "This is the home page";
-		}
-		
-		// Updating accountBalance after debit
-		
-		
-		@PostMapping("/debit/{accountNumber}/{amount}")
-		public User debit(@PathVariable String accountNumber, @PathVariable String amount) {
-			
-			User user = this.accountServices.getUser(Long.parseLong(accountNumber));
-			
-			if ( Long.parseLong(amount) <= user.getAccountBalance() ) {
-				
-				user.updateBalanceBy(-Long.parseLong(amount)) ;
-				
-				return this.accountServices.updateUser(user);
-			}
-			
-			else 
-				return user;
-		}
-		
-		// Updating accountBalance after credit
-		
-		@PostMapping("/credit/{accountNumber}/{amount}")
-		public User credit(@PathVariable String accountNumber, @PathVariable String amount) {
-				
-			User user = this.accountServices.getUser(Long.parseLong(accountNumber));
-			
-			if ( Long.parseLong(amount) <= user.getAccountBalance() ) {
-				
-				user.updateBalanceBy(Long.parseLong(amount)) ;
-				
-				return this.accountServices.updateUser(user);
-			}
-			
-			else 
-				return user;
 
-		}
+
+    @Autowired
+    private AccountService accountService;
+
+    @Autowired
+    private TransactionService transactionService;
+
+    // Updating accountBalance after debit
+
+    @PostMapping("/debit/{accountNumber}")
+    public ResponseEntity<String> debit(@PathVariable @Min(1) Long accountNumber, @RequestParam @Min(1) Long amount) throws CustomExceptions {
+        User user = accountService.getUser(accountNumber);
+        if ( amount.compareTo(user.getAccountBalance()) <= 0){
```

Comment:
such code should be in service layer not on controller layer, discuss it with peers.